components/ExtInfoxSgWidget/ExtInfoxSgWidget.tsx

```diff
@@ -0,0 +1,78 @@
+import * as React from 'react';
+import { render } from 'react-dom';
+import './ExtInfoxSgWidget.scss';
+import { ExtEmbed } from '../ExtEmbed/ExtEmbed';
+
+interface IState {
+    htmlString: string;
+}
+
+type ComponentProps = {
+    'data-script-id': string;
+    'data-height': string;
+    'data-width'?: string;
+} & Record<string, string | number | boolean>;
```

Comment:
Попробовал убрать. Не проходит lint. Я честно скопировал этот кусок кода из ExtGiraff, Smi2 и Zen - там описание аналогичное

Reply:
Ошибка, о которой репортил eslint - `Use an interface instead of a type literal  @typescript-eslint/prefer-interface`

Она говорит о том, что в случае отсутствия `& Record<string, string | number | boolean>` можно обойтись уже интерфейсом вместо типа.

Интерфейсы в-целом предпочтительнее типов, так как [обладают](https://github.com/bradzacher/eslint-plugin-typescript/blob/master/docs/rules/prefer-interface.md#prefer-an-interface-declaration-over-a-type-literal-type-t-----prefer-interface) некоторыми дополнительными свойствами.

Также при превращении типа в интерфейс нужно добавить ему префикс `I` в начале имени.

Т.е. в итоге должно получиться так:
```js
interface IComponentProps {
    'data-script-id': string;
    'data-height': string;
    'data-width'?: string;
}
```

И после этого останется только поправить места использования интерфейса.

Reply:
Исправлено